services/master/controllers/deployCheckers.go

```diff
@@ -0,0 +1,136 @@
+package controllers
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+
+	"github.com/gofiber/fiber/v2"
+	archiver "github.com/mholt/archiver/v3"
+	checker "github.com/sdslabs/katana/services/challengecheckerservice"
+)
+
+func deleteFolder(folderName string) (message string) {
+	basePath, _ := os.Getwd()
+	dirPath := basePath + "/challenges/" + folderName
+
+	// Delete the challenge folder
+	err := os.RemoveAll(dirPath)
+	if err != nil {
+		fmt.Println("Error deleting challenge folder:", err)
+		return "Error deleting challenge folder"
+	}
+
+	dirPath = basePath + "/chall/" + folderName
+	err = os.RemoveAll(dirPath)
+	if err != nil {
+		fmt.Println("Error deleting challenge folder:", err)
+		return "Error deleting challenge folder"
+	}
+	fmt.Println("Challenge folder deleted")
+	return "Challenge folder deleted"
+}
+
+func DeployChecker(c *fiber.Ctx) error {
+	folderName := ""
+	fmt.Println("Starting")
+	if form, err := c.MultipartForm(); err == nil {
+		checkerFiles := form.File["challengeChecker"]
+
+		// Loops through all challenges, if multiple uploaded :
+		for _, file := range checkerFiles {
+			pattern := `([^/]+)\.tar\.gz$`
+			regex := regexp.MustCompile(pattern)
+			match := regex.FindStringSubmatch(file.Filename)
+			folderName = match[1]
+
+			response, _ := createfolder(folderName + "-checker")
+			if response == 1 {
+				fmt.Println("Directory for challenge checker already exists with same name")
+				return c.SendString("Directory for challenge checker already exists with same name")
+			} else if response == 2 {
+				fmt.Println("Issue with creating challenge checkor directory.Check permissions")
+				return c.SendString("Issue with creating challenge checkor directory.Check permissions")
```

Comment:
Shorten the return message and I don't think so permissions would have to do anything with this?